addons/website_sale/views/templates.xml

```diff
@@ -2303,310 +2304,108 @@
-    <template id="address_kanban" name="Kanban address">
-        <t t-set="address_type" t-value="is_invoice and 'billing' or 'delivery'"/>
-        <div
-            name="address_card"
-            t-attf-class="card position-relative h-100 #{selected and 'bg-primary border border-primary' or 'js_change_address'}"
-            t-att-data-address-type="address_type"
-            t-att-data-partner-id="contact.id"
-        >
-            <div class="card-body d-flex flex-column align-items-start">
-                <t t-esc="contact" t-options="dict(widget='contact', fields=['name', 'address'], no_marker=True)"/>
-                <t t-if="contact._can_be_edited_by_current_customer(website_sale_order, address_type)">
-                    <t t-set="new_address_href" t-value="'/shop/address?address_type=' + address_type"/>
-                    <a
-                        t-att-href="new_address_href + '&amp;partner_id=' + str(contact.id)"
-                        class="js_edit_address btn btn-link p-0 mt-auto"
-                        role="button"
-                        title="Edit this address"
-                        aria-label="Edit this address">
-                        <i class="fa fa-pencil me-1"/>Edit
-                    </a>
-                </t>
-            </div>
+    <!-- /shop/address route -->
+    <template id="website_sale.address_details" inherit_id="portal.address_details" primary="True">
+        <!-- Display/Hide b2b fields based on conuntry and settings for shop address -->
+        <t name="o_b2b_fields" position="attributes">
+            <attribute name="t-if">
+                partner_sudo._display_b2b_fields()
+            </attribute>
+        </t>
+        <!-- Hide use same address from shop address form -->
+        <div name="use_same_address" position="attributes">
+            <attribute name="class">d-none</attribute>
         </div>
     </template>
 
-    <!-- /shop/address route -->
-    <template id="website_sale.address" name="Address Management">
-        <t t-call="website_sale.checkout_layout">
-            <div class="o_wsale_address_fill">
-                <div>
-                    <t t-if="not is_anonymous_cart">
-                        <t t-if="use_delivery_as_billing">
-                            <h3 class="mb-3">Your address</h3>
-                        </t>
-                        <t t-elif="address_type == 'delivery'">
-                            <h3 class="mb-3">Delivery address</h3>
-                        </t>
-                        <t t-else="">
-                            <h3 class="mb-3">Billing address</h3>
-                        </t>
-                    </t>
+    <template id="website_sale.address" name="Address Management" inherit_id="portal.address" primary="True">
+        <t t-call="portal.portal_layout" position="attributes">
+            <attribute name="t-call">website_sale.checkout_layout</attribute>
+        </t>
```

Comment:
Why do you need to inherit the portal template? IMO it would be better to have separate templates calling the same sub-template containing the form.